multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/UrisApplicationAttributeUpdater.java

```diff
@@ -24,7 +27,32 @@ public UrisApplicationAttributeUpdater(Context context, UpdateStrategy updateStr
 
     @Override
     protected boolean shouldUpdateAttribute(CloudApplication existingApplication, CloudApplicationExtended application) {
-        return !SetUtils.isEqualSet(application.getRoutes(), existingRoutes);
+        if (!SetUtils.isEqualSet(application.getRoutes(), existingRoutes)) {
+            return true;
+        }
+        return application.getRoutes()
+                          .stream()
+                          .filter(updatedRoute -> updatedRoute.getRequestedProtocol() != null)
+                          .anyMatch(updatedRoute -> doesProtocolOfTheExistingRouteDiffer(updatedRoute,
+                                                                                         getControllerClient().getApplicationGuid(existingApplication.getName())));
```

Comment:
Does the `existingApplication` not contain the GUID? An extra request per route won't scale well

Reply:
Actually, yes, not necessary